### ISSUE TITLE
Can you override TilingRule?

### ISSUE THREAD

**Issue by 3ternal** [Unity-Technologies/2d-extras#56]
I'm trying to create a sequence of tiles as part of a custom Rule Tile. Basically, I have three variants of surface tiles that only tile in order (like A -> B -> C). So I can't use random output; I have to check which tile is to the left and right. Is it possible to do this with Rule Tiles?

**Comment by alijaya** [comment]
If i'm not mistaken, currently the Rule Tiles only check 1 tiles around it to determine which tile to put. And it can not check the tiles type, it just check if that tiles is the same tile with this one or not...

I'm not sure in your case it is possible or not.
How exactly do you want to use it?
If it's still in certain limitation, it could be, but if it's more than that, I'm afraid that you need code it yourself.

**Comment by 3ternal** [comment]
Yeah, I figured it might not be possible. I've been placing the tiles by hand and that works well enough.

For my use case, I needed to know what type of tile each Rule Tile is surrounded by. Then I could write the rest of the code myself. The goal is to create a "sequence tile": A -> B -> C -> A -> B -> C, and so on. I guess this isn't possible with the current Rule Tile implementation, since it only checks against itself and can't distinguish between different sprites, so there would be no way to know whether the tile to the right or left was a Grass A, Grass B, or Grass C.

Anyway, if there's no easy way to do this with the current implementation, we can go ahead and close this issue. Placing tiles individually works well enough for my workflow.
